apiserver/facades/agent/uniter/newlxdprofile_test.go

```diff
@@ -71,9 +73,14 @@ func (s *newLxdProfileSuite) TestWatchInstanceData(c *gc.C) {
 
 func (s *newLxdProfileSuite) TestLXDProfileName(c *gc.C) {
 	defer s.setupMocks(c).Finish()
-	s.expectUnitAndMachine()
+	s.expectUnitAndMachine(2)
```

Comment:
Why did this end up being called twice?

Reply:
Because we now retrieve the assigned machine id for the unit on the entities loop (before it was done later down because we needed the full legacy machine). So even though we have 2 errors in this test, one of the errors happens *after* we retrieved the assigned machine ID.